File: server/core/src/main/java/dev/slimevr/config/BridgeConfig.java
package dev.slimevr.config;

import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import com.fasterxml.jackson.databind.ser.std.StdKeySerializers;
import dev.slimevr.config.serializers.BooleanMapDeserializer;
import dev.slimevr.tracking.trackers.TrackerRole;
import java.util.HashMap;
import java.util.Map;


public class BridgeConfig {

	@JsonDeserialize(using = BooleanMapDeserializer.class)
	@JsonSerialize(keyUsing = StdKeySerializers.StringKeySerializer.class)
	public Map<String, Boolean> trackers = new HashMap<>();
	public boolean automaticSharedTrackersToggling = true;

	public BridgeConfig() {
	}

	public boolean getBridgeTrackerRole(TrackerRole role, boolean def) {
		return trackers.getOrDefault(role.name().toLowerCase(), def);
	}

	public void setBridgeTrackerRole(TrackerRole role, boolean val) {
		this.trackers.put(role.name().toLowerCase(), val);
	}

	public Map<String, Boolean> getTrackers() {
		return trackers;
	}
}

File: server/core/src/main/java/dev/slimevr/config/BridgeConfig.kt
package dev.slimevr.config

import dev.slimevr.tracking.trackers.TrackerRole
import kotlinx.serialization.Serializable
import java.util.*

@Serializable
class BridgeConfig {
	var trackers: MutableMap<String, Boolean> = HashMap<String, Boolean>()
	var automaticSharedTrackersToggling: Boolean = true

	fun getBridgeTrackerRole(role: TrackerRole, def: Boolean): Boolean = trackers.getOrDefault(role.name.lowercase(Locale.getDefault()), def)!!

	fun setBridgeTrackerRole(role: TrackerRole, `val`: Boolean) {
		this.trackers[role.name.lowercase(Locale.getDefault())] = `val`
	}
}
